packages/quick_actions/quick_actions_android/lib/quick_actions_android.dart

```diff
@@ -3,54 +3,73 @@
 // found in the LICENSE file.
 
 import 'package:flutter/foundation.dart';
-import 'package:flutter/services.dart';
 import 'package:quick_actions_platform_interface/quick_actions_platform_interface.dart';
 
+import 'src/messages.g.dart';
+
 export 'package:quick_actions_platform_interface/types/types.dart';
 
-const MethodChannel _channel =
-    MethodChannel('plugins.flutter.io/quick_actions_android');
+late QuickActionHandler _handler;
 
-/// An implementation of [QuickActionsPlatform] that for Android.
+/// An implementation of [QuickActionsPlatform] for Android.
 class QuickActionsAndroid extends QuickActionsPlatform {
+  /// Creates a new plugin implementation instance.
+  QuickActionsAndroid({
+    @visibleForTesting AndroidQuickActionsApi? api,
+  }) : _hostApi = api ?? AndroidQuickActionsApi();
+
+  final AndroidQuickActionsApi _hostApi;
+
   /// Registers this class as the default instance of [QuickActionsPlatform].
   static void registerWith() {
     QuickActionsPlatform.instance = QuickActionsAndroid();
   }
 
-  /// The MethodChannel that is being used by this implementation of the plugin.
-  @visibleForTesting
-  MethodChannel get channel => _channel;
-
   @override
   Future<void> initialize(QuickActionHandler handler) async {
-    channel.setMethodCallHandler((MethodCall call) async {
-      assert(call.method == 'launch');
-      handler(call.arguments as String);
-    });
-    final String? action =
-        await channel.invokeMethod<String?>('getLaunchAction');
+    _handler = handler;
+    final String? action = await _hostApi.getLaunchAction();
     if (action != null) {
-      handler(action);
+      _handler(action);
     }
+    final _QuickActionHandlerApi quickActionsHandlerApi =
+        _QuickActionHandlerApi();
+    AndroidQuickActionsFlutterApi.setup(quickActionsHandlerApi);
```

Comment:
Nit: I would keep the original order (set up the channel, then do the logic to get any queued action) since it seems more logical to do setup tasks first.